iis-wf/iis-wf-export-actionmanager/src/test/java/eu/dnetlib/iis/wf/export/actionmanager/sequencefile/TestingConsumer.java

```diff
@@ -0,0 +1,120 @@
+package eu.dnetlib.iis.wf.export.actionmanager.sequencefile;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Properties;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.Text;
+import org.apache.log4j.Logger;
+
+import eu.dnetlib.actionmanager.actions.AtomicAction;
+import eu.dnetlib.iis.common.java.PortBindings;
+import eu.dnetlib.iis.common.java.Process;
+import eu.dnetlib.iis.common.java.io.FileSystemPath;
+import eu.dnetlib.iis.common.java.io.SequenceFileTextValueReader;
+import eu.dnetlib.iis.common.java.porttype.AnyPortType;
+import eu.dnetlib.iis.common.java.porttype.PortType;
+
+/**
+ * Sequence file testing consumer. Expects {@link Text} values at input.
+ * Requirements are provided as comma separated list of properties files 
+ * holding requirements for each individual input object. 
+ * Properties keys are paths to the fields we want to test in given object, 
+ * properties values are expected values (with special markup for null: $NULL$).
+ * This means number of input records in sequence file should be equal to the number of provided properties files.
+ *   
+ * @author mhorst
+ */
+public class TestingConsumer implements Process {
+
+	public static final String PROPERTIES_CSV = "expectation_properties_csv";
+
+	public static final String PORT_INPUT = "seqfile";
+
+	public static final String NULL_VALUE_INDICATOR = "$NULL$";
+
+	private final FieldAccessor accessor;
+	
+	private final static Logger log = Logger.getLogger(TestingConsumer.class);
+
+	private static final Map<String, PortType> inputPorts = new HashMap<String, PortType>();
+	
+	{
+		inputPorts.put(PORT_INPUT, new AnyPortType());
+	}
+	
+	public TestingConsumer() {
+		accessor = new FieldAccessor();
+		accessor.registerDecoder("TargetValue", new OafFieldDecoder());
+	}
+
+	@Override
+	public Map<String, PortType> getInputPorts() {
+		return inputPorts;
+	}
+
+	@Override
+	public Map<String, PortType> getOutputPorts() {
+		return new HashMap<String, PortType>();
+	}
+
+	@Override
+	public void run(PortBindings portBindings, Configuration configuration, Map<String, String> parameters)
+			throws Exception {
+		String propertiesPathsCSV = parameters.get(PROPERTIES_CSV);
+		Path inputPath = portBindings.getInput().get(PORT_INPUT);
+		if (StringUtils.isEmpty(propertiesPathsCSV)) {
+			throw new Exception("no " + PROPERTIES_CSV + " property value provided, "
+					+ "output requirements were not specified!");
+		}
+		String[] recordsSpecs = StringUtils.split(propertiesPathsCSV, ',');
+		FileSystem fs = FileSystem.get(configuration);
+		if (fs.exists(inputPath)) {
```

Comment:
Throw if not exists, fall through to the happy path otherwise. This puts precondition handling first and avoids some indentation.